### PR TITLE
Fix to add clustering_callbacks.py to the pip package

### PR DESCRIPTION
**Pull request by benkli01**
This is a small fix to add a dependency on `clustering_callbacks.py` so that the file is added to the pip package.

### CODE REVIEW

**Review by akarmi — APPROVED** [review]
Looks OK to me, but @alanchiao, please have a look if this is a correct way of instructing bazel to include this module into the wheel.

**Comment by alanchiao** [comment]
@benkli01 

Please refer to an example like https://github.com/tensorflow/model-optimization/pull/421 for adding changes to the pip package. You can subsequently install from source and test the actual path like in https://www.tensorflow.org/model_optimization/guide/install#installing_from_source.

**Comment by alanchiao** [comment]
Please also refer to https://github.com/tensorflow/model-optimization/blob/master/CONTRIBUTING.md#style-and-practices in updating the release notes given this feature.

**Comment by benkli01** [comment]
Hi @alanchiao , thanks for the feedback. 

- Regarding the release notes: Is the RELEASE.md in branch **master** ready to be edited? Should I just overwrite the template or add a new section for the next release? 
- I have added the **ClusteringSummaries** to the clustering API as well (not only to the BUILD file).

**Comment by alanchiao** [comment]
Yes the release notes are ready to be edited. You can add a new section for the next release with a TBD release version.

**Comment by alanchiao** [comment]
Changes otherwise look good. 

To now actually do the API review since the previous [PR](https://github.com/tensorflow/model-optimization/pull/508/files) didn't expose anything: it'd be good to:

a) Include a screenshot of a sample of what you'd see on Tensorboard from these summaries
b) Just as a note for others: ClusteringSummaries sets a default of 'logs' for log_dir, which matches the [TensorBoard callback](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/TensorBoard) but differs from [PruningSummaries](https://www.tensorflow.org/model_optimization/api_docs/python/tfmot/sparsity/keras/PruningSummaries). Really, it makes sense to do what was
done for ClusteringSummaries.
c) Is there a reason to have a separate `cluster_update_freq`from `update_freq`? Really, it's probably best of PruningSummaries didn't inherent from the TensorBoard callback - it's confusing since if a user uses both PruningSummaries and TensorBoard callbacks, then the TensorBoard callback work would be done twice.

**Comment by alanchiao** [comment]
We need to otherwise get a look by either @nutsiepully or @liyunlu0618.

File: RELEASE.md [hunk 1]
```diff
@@ -30,6 +30,14 @@ Keras pruning API:
 * Tested against TensorFlow 1.14.0, 2.0.0, and nightly, and Python 3.
 
 
+# TensorFlow Model Optimization next release TBD
+
+Keras clustering API:
+
+* Added *ClusteringSummaries* to create additional output for the clustering 
+progress for TensorBoard.
+* Tested against TensorFlow 1.14.0, 2.0.0, and nightly, and Python 3.
```
Comment (wwwind) [inline RELEASE.md:39]: nit: 2.3.0, in 0.5.0 we are talking about the latest versions

Comment (alanchiao) [reply RELEASE.md:39]: Actually 2.0.0 is fine and intended. 

The idea is that if it passes against 2.0.0 and nightly, then it should generally pass on all versions in between (including 2.1 / 2.2) given that 2.X has no backward compatible changes. It won't be perfect (e.g. miscellaneous bugs), but it's the most practical choice given we cannot test across all versions.

Comment (alanchiao) [reply RELEASE.md:39]: And until we communicate otherwise, currently TFMOT is expected to be compatible with 2.0+.

**Comment by alanchiao** [comment]
@benkli01: could you comment on "is there a reason to have a separate cluster_update_freq from update_freq? Really, it's probably best of PruningSummaries didn't inherent from the TensorBoard callback - it's confusing since if a user uses both PruningSummaries and TensorBoard callbacks, then the TensorBoard callback work would be done twice". It's the main thing that stood out to me in this API.

**Comment by benkli01** [comment]
> 
> 
> @benkli01: could you comment on "is there a reason to have a separate cluster_update_freq from update_freq? Really, it's probably best of PruningSummaries didn't inherent from the TensorBoard callback - it's confusing since if a user uses both PruningSummaries and TensorBoard callbacks, then the TensorBoard callback work would be done twice". It's the main thing that stood out to me in this API.

Hi @alanchiao , I am really sorry for the delay. There were some internal discussions about this PR.
The callback is derived from the TensorBoard callback because it produces very consistent summaries and the PruningSummaries callback does it as well, so I thought it would be best to be consistent, but I understand your point. From the user perspective _adding_ another callback might be more intuitive than _replacing_ the TensorBoard callback. It might lead to problems/inconsistencies when writing the summaries to the disk though, but I would have to further investigate that.

File: RELEASE.md (hunk 1 above)
Comment (benkli01) [reply RELEASE.md:39]: Thanks for the clarification, Alan!

**Comment by alanchiao** [comment]
Thanks Benjamin for the response. I think it's fine to be consistent with inheriting from the callback for now given it's what Pruning does.

I actually intended to highlight the part ""is there a reason to have a separate cluster_update_freq from update_freq" - is it valuable to have the two separate? 
I accidentally copied more of the quote than I should have.

**Comment by benkli01** [comment]
> 
> 
> I actually intended to highlight the part ""is there a reason to have a separate cluster_update_freq from update_freq" - is it valuable to have the two separate?

This is actually the main addition of the ClusteringSummaries. It allows to save histograms of the clustered weights per batch instead of just per epoch, so that the clustering can be observed at a finer scale. To be honest, after thinking about it I'm not sure if that is enough to justify the new class.

**Review by wwwind — APPROVED** [review]

**Review by daverim — APPROVED** [review]
LGTM

**Pull request merged**